src/main/java/eu/xenit/gradle/docker/compose/DockerComposeConventionImpl.java

```diff
@@ -108,23 +129,32 @@ public void fromProject(Project project) {
                 composeTask.doFirst(CONFIGURE_COMPOSE_ACTION_NAME, new Action<Task>() { // No lambda -> see #96
                     @Override
                     public void execute(Task t) {
-                        createSetComposeEnvironmentFromPathAction().execute(dockerBuildImage);
+                        if(environmentVariable != null) {
+                            String addition = "_" + dockerBuildImage.getName();
+                            String safeEnvironmentVariable = Util.safeEnvironmentVariableName(environmentVariable + addition);
+                            System.out.println("SAFE: " + safeEnvironmentVariable);
+                            createSetComposeEnvironmentAction(safeEnvironmentVariable).execute(dockerBuildImage);
+                        }else{
+                            createSetComposeEnvironmentFromPathAction().execute(dockerBuildImage);
+                        }
                     }
                 });
             });
         });
 
         // Register shortened environment variables for `buildDockerImage` tasks created with the docker plugin
         pluginClasspathChecker.withPlugin(project, DockerPlugin.class, DockerPlugin.PLUGIN_ID, plugin -> {
-            String environmentName = Util.safeEnvironmentVariableName(project.getPath().substring(1)) + "_DOCKER_IMAGE";
-            fromBuildImage(environmentName, project.getTasks().named("buildDockerImage", DockerBuildImage.class));
+            if(environmentVariable != null) {
+                System.out.println("S: " + Util.safeEnvironmentVariableName(environmentVariable));
+                fromBuildImage(Util.safeEnvironmentVariableName(environmentVariable),
```

Comment:
The user-provided part of the environment name should not be passed to `Util.saveEnvironmentVariableName()`, it is the responsibility of the caller to ensure that the environment variable name is correct.